components/postgresql/postgresql.app.mjs

```diff
@@ -1,11 +1,215 @@
+    },
+    async endClient(client) {
+      return client.end();
+    },
+    /**
+     * Executes SQL query and returns the resulting rows
+     * @param {string} text - SQL query to execute
+     * @param {array} [values] - values to substitute into the given query
+     * @returns Array of rows returned from the given SQL query
+     */
+    async executeQuery(text, values = null) {
+      const client = await this.getClient();
+      const query = values
+        ? {
+          text,
+          values,
+        }
+        : text;
+      try {
+        const { rows } = await client.query(query);
+        return rows;
+      } finally {
+        await this.endClient(client);
+      }
+    },
+    /**
+     * Gets an array of table names in a database
+     * @returns Array of table names
+     */
+    async getTables() {
+      const rows = await this.executeQuery("SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'");
+      return rows.map((row) => row.table_name);
+    },
+    /**
+     * Gets an array of column names in a table
+     * @param {string} table - Name of the table to get columns in
+     * @returns Array of column names
+     */
+    async getColumns(table) {
+      const rows = await this.executeQuery(`SELECT column_name FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_NAME = '${table}'`);
+      return rows.map((row) => row.column_name);
+    },
+    /**
+     * Gets a table's primary key field
+     * @param {string} table - Name of the table to get the primary key for
+     * @returns Name of the primary key column
+     */
+    async getPrimaryKey(table) {
+      const rows = await this.executeQuery(`SELECT a.attname FROM pg_index i JOIN pg_attribute a ON a.attrelid = i.indrelid AND a.attnum = ANY(i.indkey) WHERE i.indrelid = '${table}'::regclass AND i.indisprimary`);
+      return rows[0].attname;
+    },
+    /**
+     * Gets rows in a table with values greater than lastResult
+     * @param {string} table - Name of database table to query
+     * @param {string} column - Column to filter by lastResult
+     * @param {string} [lastResult] - A column value. Used to retrieve new rows since the
+     *  last time the table was queried.
+     * @returns Array of rows returned from the query
+     */
+    async getRows(table, column, lastResult = null) {
+      const query = `SELECT * FROM ${table} ${lastResult
+        ? "WHERE " + column + ">" + lastResult
+        : ""} ORDER BY ${column} DESC`;
+      return this.executeQuery(query);
+    },
+    /**
+     * Gets a single row from a table based on a lookup column & value
+     * @param {string} table - Name of database table to query
+     * @param {string} column - Column to filter by value
+     * @param {string} value - A column value to search for
+     * @returns A single database row
+     */
+    async findRowByValue(table, column, value) {
+      const rows = await this.executeQuery(`SELECT * FROM ${table} WHERE ${column} = '${value}'`);
```

Comment:
If you change `executeQuery` the way I suggested previously I would refactor this one to
```suggestion
      const rows = await this.executeQuery({
        text: format("SELECT * FROM %I WHERE %I = $1", table, column),
        values: [
          value,
        ],
      });
```